src/lib/utils/classNames.js

```diff
@@ -0,0 +1,8 @@
+export default (...classNames) => {
+  const filteredClassNames = classNames.filter(
```

Comment:
I'm not sure if this test is maybe too strict.

I was thinking more along the lines of 
```js
 const filteredClassNames = classNames.filter((className) =>  !!className);
```

Apart the unlikely case of having classes with numeric names the only argument against your solution I can think of is speed. I run a quick test in chrome:

```js
var iterations = 1000000;
console.time('Function #1');
for(var i = 0; i < iterations; i++ ){
    const test = (className) => typeof className === 'string' && className.trim().length > 0;

    test('');
    test(false);
    test(1);
    test('name');
};
console.timeEnd('Function #1')

console.time('Function #2');
for(var i = 0; i < iterations; i++ ){
    const test = (className) => !!className;

    test('');
    test(false);
    test(1);
    test('name');
};
console.timeEnd('Function #2')
VM19564:11 Function #1: 23.919189453125 ms
VM19564:22 Function #2: 5.922119140625 ms
```

So `!!` is somewhat faster, but both are pretty fast.

I would probably use `!!` but I do not feel it too strongly about it.

Reply:
Agree, it is faster. The idea has been to allow only non-empty string so that `null`, `undefined`, `<bool>`, `<number>`, ... are filtered out. Same for empty string which should not be part of class names as it creates multiple empty spaces which we have wanted to omit. And in that case there is no faster way than `typeof className === 'string' && className.trim().length > 0` check. But if we agree that we are that responsible that we do not use empty strings in conditions (I hope I removed all of them) then we can omit that.

Reply:
Please:
* move `src/lib/helpers` to `src/lib/components/_helpers`
* move `src/lib/hooks` to `src/lib/components/ScrollView/hooks`
* move `src/lib/helpers/getElementsPositionDifference` to `src/lib/components/ScrollView/_helpers`

Reply:
@adamkudrna has no idea what you are talking about. But he thinks our most frequented use case `styles.*` is somehow omitted here… (although it works!)

Reply:
I have already made changes in those folders and files. But I still haven't changed the detection using typeof. Should I change it or not? I would like to have this type of change instead of what Martin wrote.

Reply:
If I get it correctly, we agreed on keeping `typeof` today.